ptrail/regularization/goemetric_distance_geometry.py

```diff
@@ -0,0 +1,237 @@
+"""
+    | Author: Abhishek Gujjar
+"""
+
+import pandas as pd
+import numpy as np
+from typing import Union, List
+
+
+def traj_rediscretize_wrapper(trj, step) -> Union[np.ndarray, None]:
+    """
+    Wrapper function to handle trajectory rediscretization with error handling.
+
+    Args:
+        trj: Input trajectory
+        step: Rediscretization step
+
+    Returns:
+        Rediscretized trajectory or None if an exception occurs.
+    """
+    try:
+        return TrajRediscretize(trj, step)
+    except:
+        return None
+
+
+def traj_rediscretize_wrapper_default(trj, step) -> np.ndarray:
+    """
+    Wrapper function to handle trajectory rediscretization with a default step value if an error occurs.
+
+    Args:
+        trj: Input trajectory
+        step: Rediscretization step
+
+    Returns:
+        Rediscretized trajectory using the provided step or a default step value if an exception occurs.
+    """
+    try:
+        return TrajRediscretize(trj, step)
```

Comment:
These seem like built in functions of an R-package called trajr. See: https://rdrr.io/cran/trajr/. Similar functions are being used in several other files, find an equivalent for those and replace them, these calls are errors otherwise. A good substitute can be this: https://github.com/traja-team/traja. Please evaluate and reach out back so we can discuss.